File: JAVA/Product.java
/*
    Saya Revana Faliha Salma NIM 2202869 mengerjakan Soal Latihan 2 
    dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak 
    melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/

public class Product{
    private String id_Product;  // deklarasi variabel untuk menyimpan ID produk
    private String name;    // deklarasi variabel untuk menyimpan nama produk
    private String brand;   // deklarasi variabel untuk menyimpan merek produk
    private int price;  // deklarasi variabel untuk menyimpan harga produk

    // konstruktor kosong
    Product(){

    }

    // konstruktor parameter untuk menginisialisasi objek Product dengan nilai yang diberikan
    public Product(String id, String nama, String brand, int harga){
        this.id_Product = id;   // inisialisasi ID produk
        this.name = nama;   // inisialisasi nama produk
        this.brand = brand; // inisialisasi merek produk
        this.price = harga; // inisialisasi harga produk
    }

    // metode setter untuk mengatur ID produk
    public void setId(String id){
        this.id_Product = id;
    }

    // metode getter untuk mendapatkan ID produk
    public String getId(){
        return this.id_Product;
    }
    
    // metode setter untuk mengatur nama produk
    public void setName(String name){
        this.name = name;
    }

    // metode getter untuk mendapatkan nama produk
    public String getName(){
        return this.name;
    }

    // metode setter untuk mengatur merek produk
    public void setBrand(String brand){
        this.brand = brand;
    }

    // metode getter untuk mendapatkan merek produk
    public String getBrand(){
        return this.brand;
    }

    // metode setter untuk mengatur harga produk
    public void setPrice(int price){
        this.price = price;
    }

    // metode getter untuk mendapatkan harga produk
    public int getPrice(){
        return this.price;
    }
}

File: PYTHON/Program/Product.py
# Saya Revana Faliha Salma NIM 2202869 mengerjakan Soal Latihan 2 
# dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak 
# melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

class Product:
    
    __idProduct = ""    # variabel privat untuk ID produk
    __name = ""     # variabel privat untuk nama produk
    __brand = ""    # variabel privat untuk merek produk
    __price = ""    # variabel privat untuk harga produk

    # konstruktor dengan parameter opsional untuk setiap atribut
    def __init__(self, id = "", nama = "", brand = "", price = ""):
        self.__id = id  # menginisialisasi atribut ID produk
        self.__nama = nama   # menginisialisasi atribut nama produk
        self.__brand = brand    # menginisialisasi atribut merek produk
        self.__price = price    # menginisialisasi atribut harga produk

    # metode untuk mengatur ID produk
    def set_id(self, id):
        self.__idProduct = id

    # metode untuk mendapatkan ID produk
    def get_id(self):
        return self.__idProduct
    
    # metode untuk mengatur nama produk
    def set_name(self, name):
        self.__name = name

    # metode untuk mendapatkan nama produk
    def get_name(self):
        return self.__name
    
    # metode untuk mengatur merek produk
    def set_brand(self, brand):
        self.__brand = brand

    # metode untuk mendapatkan merek produk
    def get_brand(self):
        return self.__brand
    
    # metode untuk mengatur harga produk
    def set_price(self, price):
        self.__price = price

    # metode untuk mendapatkan harga produk
    def get_price(self):
        return self.__price
